src/resolvers/api/getNode.js

```diff
@@ -0,0 +1,147 @@
+const _get = require('lodash.get')
+
+const { query } = require('../utils')
+
+const getNode = async (_, args) => {
+  const { urbitId } = _get(args, 'input.urbitId') || null
+
+  let nodeType = null
+
+  if (urbitId.length === 4) {
+    nodeType = 'GALAXY'
+  } else if (urbitId.length === 7) {
+    nodeType = 'STAR'
+  } else if (urbitId.length === 14) {
+    nodeType = 'PLANET'
+  }
+
+  const getNumOwnersQuery = `select count(*) from pki_events where node_id = '${urbitId}' and event_type_id = '1';`
+  const getNumOwnersResponse = await query(getNumOwnersQuery)
+
+  const numOwners = parseInt(_get(getNumOwnersResponse, 'rows[0].count')) || 1
+
+  const sponsors = []
+
+  // sponsor
+  const getSponsorQuery = `select sponsor_id from pki_events where node_id = '${urbitId}' order by time desc limit 1;`
+  const getSponsorResponse = await query(getSponsorQuery)
+  const getSponsorResponseRow = _get(getSponsorResponse, 'rows[0]') || []
+
+  const sponsor = getSponsorResponseRow ? _get(getSponsorResponseRow, 'sponsor_id') || null : null
+
+  sponsors.push(sponsor)
+
+  // sponsor's sponsor
+  const getSponsorsSponsorQuery = `select sponsor_id from pki_events where node_id = '${sponsor}' order by time desc limit 1;`
+  const getSponsorsSponsorResponse = await query(getSponsorsSponsorQuery)
+  const getSponsorsSponsorResponseRow = _get(getSponsorsSponsorResponse, 'rows[0]') || []
+  const sponsorsSponsor = getSponsorsSponsorResponseRow ? _get(getSponsorsSponsorResponseRow, 'sponsor_id') || null : null
+
+  if (sponsorsSponsor) {
+    sponsors.push(sponsorsSponsor)
+  }
+
+  // status
```

Comment:
This whole status calculation is pretty messy. The nesting is unnecessary and makes things hard to read, as does the rebinding of `getStatusId{Query,Response}`. I'd prefer it if you:

- Broke this out into its own function that, given an `urbitId`, produces the appropriate status.
- Since you're now working with a function, you can `return` a status once you find it.
- If you start with the innermost query (with the hard-coded addresses), you'll know whether or not you need to proceed with subsequent queries. Happy to talk through this during one of our sessions.

Reply:
Good call. Definitely makes the file look a lot better to abstract that out, and of course any time one avoids nested conditionals is good. I moved the function to `/network-api/src/resolvers/utils.js`. Let me know if you'd prefer it somewhere else.